home/views/add_professor.py

```diff
@@ -47,3 +46,56 @@ def post(self, request):
             }
 
         return JsonResponse(context)
+
+class EditReview(View):
+    def post(self, request):
+        user = request.user
+        review = Review.unfiltered.filter(pk=request.POST['review_id']).select_related("course").first()
+        initial = {
+            'content': review.content,
+            'course': review.course.name if review.course else None,
+            'rating': review.rating,
+            'grade': review.grade,
+            'anonymous': review.anonymous,
+            "review_id": review.pk
+        }
+        form = EditReviewForm(user, data=request.POST, initial=initial)
+
+        if form.is_valid():
+            if 'content' in form.changed_data:
+                review.status = Review.Status.PENDING
+                review.content = form.cleaned_data['content']
+
+            course = form.cleaned_data['course']
+            if course != '' and 'course' in form.changed_data:
+                new_course = Course.unfiltered.filter(name=course).first()
```

Comment:
Commenting on https://github.com/planetterp/PlanetTerp/pull/67/files/36f9117760235bfcec8f766670f0da38f16ffc16#r1137569447 here:

I don't think `.get()` would be appropriate here because duplicate courses is a real problem, which would cause a `MultipleObjectsReturned` error